### ISSUE TITLE
(disambiguation) k9 should notify when user refresh and there is no network connection

### ISSUE THREAD

**Issue by ReleaseStandard** [k9mail/k-9#5425]
**Is your feature request related to a problem? Please describe.**
When you got no network and swipe down to refresh, the spinner quickly appear and disappear.
From that point user can believe that:
 1: there is no new mails
 2: there is an issue in k9
 3: there is a network issue
most basic users will fall into 1: then 2:

**Describe the solution you'd like**
Show disambiguation information, e.g. show toast notification "Could not fetch emails, there is no network connection".

**Describe alternatives you've considered**
Show a notification.
Show a banner (e.g. red banner) no network connection.

**Additional context**

**Comment by ReleaseStandard** [comment]
Maybe adding a preference also to tweak this behaviour ?

**Comment by cketti** [comment]
I think for manually triggered syncs displaying a snackbar with an error message would be nice.

We have to be a bit careful with the error message. Just because fetching messages failed doesn't mean there's no network connectivity.

**Comment by ReleaseStandard** [comment]
Yeah displaying messages in snackbar could be an option but require some more implied member in k9 dev than me (I'm just tweaking an existing project).

#5426 add feedback not only for network error (it was just an example).
It shows to user (as notification) to the cause of the fetch mail error examples below :
 1. notification title:
              Checking mail failed 
   notification content:
              Unable to resolve host "mail.mailo.com": No address associated with hostname.
 2. notification title:
              Checking mail failed
  notification content:
              Command ... Authenticate error

Issue here is that 2. raise notification on authenticate error but there is already a notifcation to do so.

Solution to that could be : 
I think for manually triggered syncs displaying a snackbar with an error message would be nice.
so below action must display feedback (I'm only considering notifications as snackbar notifications could be more difficult to implement) in case of fetch error:
 - load more messages
 - swipe down to refresh mails
 - refresh folder list from the overflow menu

**Comment by gnued** [comment]
Related: please give error message whenever mail cannot be fetched. Eg account blocked, network error, password compromised. At present K9 fails silently and behaves exactly as if no new mail.

**Comment by ReleaseStandard** [comment]
That precisly the point there is no error feedback
 https://we.tl/t-pPQXFuChyv. Animation play as if mails have been correctly fetched. But that's not the case  because there is not network).

**Comment by fadainasch** [comment]
I would like to suggest implementing a similar behavior for manual email fetching in the inbox (by swiping down), as the current server search function in the app already displays a prompt "A network connection is required for server search" and later "Remote search failed" when a network connection is unavailable.
